src/init.c

```diff
@@ -53,6 +55,27 @@ extern int __data_start;
 extern int _end;
 #endif
 
+#ifdef __APPLE__
+vm_address_t get_base_address(mach_port_t task, size_t *region_size, void *addr)
+{
+    kern_return_t kret;
+    vm_region_basic_info_data_t info;
+    vm_size_t size;
+    mach_port_t object_name;
+    mach_msg_type_number_t count;
+    mach_vm_address_t address = (addr ? (mach_vm_address_t)addr : 0);
+    count = VM_REGION_BASIC_INFO_COUNT_64;
+
+    kret = mach_vm_region(task, &address, (mach_vm_size_t *)&size,
+                          VM_REGION_BASIC_INFO, (vm_region_info_t)&info,
+                          &count, &object_name);
+    assert(kret == KERN_SUCCESS);
```

Comment:
I am trusting your experience with this change as I have little idea about it. One question, if the `assert` fails here, does that tell the user exactly why the program aborted?

Reply:
I agree with Wasi. I like the `assert`, it might be nice to include a message for the user if the program were to abort here.

Reply:
Here's what the assertion prints when things go wrong:
```
Assertion failed: (kret != KERN_SUCCESS), function get_base_address, file ../../src/init.c, line 72.
```
Do you think that's sufficient?

Reply:
`shmem_internal_assertp` may be used as well.

Reply:
This is `shmem_internal_assertp`:
```
[0000] Assertion Failed: ../../src/init.c:72: get_base_address
[0000]                   kret != KERN_SUCCESS
```

Preference?  This option does have the PE number...

Reply:
Looks to have a minimal impact. Leaving it to you. Either way, it is ok.

Reply:
I went with `shmem_internal_assertp` - does it look good?

If so, is the PR ready to merge?

Reply:
Looks good. Please feel free to merge.